snap/snapcraft.yaml

```diff
@@ -1,17 +1,96 @@
-name: my-snap-name # you probably want to 'snapcraft register <name>'
-base: core24 # the base snap is the execution environment for this snap
-version: '0.1' # just for humans, typically '1.2+git' or '1.3.2'
-summary: Single-line elevator pitch for your amazing snap # 79 char long summary
+name: intel-npu-driver
+base: core22
+version: '1.6.0'
+summary: User Mode Driver with Compiler Software for the Intel® NPU
 description: |
-  This is my-snap's description. You have a paragraph or two to tell the
-  most important story about your snap. Keep it under 100 words though,
-  we live in tweetspace and your description wants to look good in the snap
-  store.
+  User mode driver (UMD) with compiler for the Intel® NPU software stack.
+  Intel® NPU device is an AI inference accelerator integrated with Intel
+  client CPUs, starting from Intel® Core™ Ultra generation of CPUs (formerly
+  known as Meteor Lake). It enables energy-efficient execution of artificial
+  neural network tasks. This snap serves primarily as a producer snap providing
+  the oneAPI Level-Zero and UMD with compiler libraries. It also distributes
+  apps for installing firmware to a NPU device and validating the UMD.
+  For more info, see https://github.com/intel/linux-npu-driver
+grade: stable
+confinement: strict
 
-grade: devel # must be 'stable' to release into candidate/stable channels
-confinement: devmode # use 'strict' once you have the right plugs and slots
+slots:
+  intel-npu:
+    interface: custom-device
+    custom-device: intel-npu-device
+    devices:
+      - /dev/accel/accel[0-9]
+      - /dev/accel/accel[1-5][0-9]
+      - /dev/accel/accel6[0-3]
+    udev-tagging:
+      - kernel: accel[0-9]
+        subsystem: accel
+        attributes:
+          vendor: "0x8086" # Intel
+      - kernel: accel[1-5][0-9]
+        subsystem: accel
+        attributes:
+          vendor: "0x8086" # Intel
+      - kernel: accel6[0-3]
+        subsystem: accel
+        attributes:
+          vendor: "0x8086" # Intel
+  npu-libs:
+    interface: content
+    content: npu-libs-2204
+    read:
+      - $SNAP/usr/lib/x86_64-linux-gnu
+
+plugs:
+  intel-npu-plug: # append -plug as plug name cannot match slot name in same snap
+    interface: custom-device
+    custom-device: intel-npu-device
+  intel-npu-fw:
+    interface: kernel-firmware-control
+
+hooks:
+  connect-plug-intel-npu-fw:
+    plugs: [intel-npu-fw]
+  disconnect-plug-intel-npu-fw:
```

Comment:
Out of curiosity, is that hook also invoked at snap removal ?

Reply:
Yeah, when you `sudo snap remove intel-npu-driver` that first triggers the remove hook and then the `disconnect-plug-intel-npu-fw` hook (I guess that is the normal flow, i.e. all of a snap's active connections are disconnected before its removed).

A few related notes for clarity:

* This snap also uses a `install` and [`remove` hook](https://github.com/canonical/intel-npu-driver-snap/blob/frenchwr/build-producer-snap/snap/hooks/remove). Those aren't declared explicitly in the `snapcraft.yaml` because they do not need any special interfaces.
* I tested just to make sure that when the snap is removed the custom firmware path is removed from `/sys/module/firmware_class/parameters/path` via the `disconnect-plug-intel-npu-fw` hook.